pkg/gofr/datasource/pubsub/mqtt/mqtt.go

```diff
@@ -318,7 +319,15 @@ func getHandler(subscribeFunc SubscribeFunc) func(client mqtt.Client, msg mqtt.M
 		}
 
 		// call the user defined function
-		_ = subscribeFunc(pubsubMsg)
+		err := subscribeFunc(pubsubMsg)
+		if err != nil {
+			m.logger.Errorf("error subscribing to message, err : %v", err)
+
+			return
+		}
+
+		// acknowledge messge if subscription is successful
+		msg.Ack()
```

Comment:
There's another way to ack messages now. It's done one level above in subscription manager using Commiter interface (see mqtt/message.go)
Commit already made here
https://github.com/gofr-dev/gofr/blob/63ebe9e9a6dc9162f36eb5c5792cda0d71e873dd/pkg/gofr/subscriber.go#L52

Reply:
The function is `SubscribeWithFunc`, an extended version of `Subscribe`. It is not called from the subscription manager, so the messages are not acknowledged, but this made me realise that the `pubsub.Message` created here was not populating the Commiter interface.

Reply:
I'll try to debug later for better understanding what happens